specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts

```diff
@@ -304,18 +287,17 @@ export class BusinessRuleManager<SCHEMA extends AnySchema> {
 
       return fieldValues.reduce(
         (previous, current, index) =>
-          previous && hasSameValue(current, fieldNames![index]),
+          previous && hasSameValue(current, fieldNames[index]),
         true
       );
     };
 
     // If the uniqueness rule should be unique to database
-    if (scope === undefined) {
+    if (scopes.length === 0) {
       const filters: Partial<IR<boolean | number | string | null>> = {};
-
-      for (const [f, fieldName] of fieldNames.entries()) {
-        filters[fieldName] = fieldIds[f] || fieldValues[f];
-      }
+      fieldNames.forEach((field, index) => {
+        filters[field] = fieldIds[index] ?? fieldValues[index];
+      });
```

Comment:
Please make trivial code changes along the way
```
const filters = Object.fromEntries(fieldNames.map(
(fieldName, index)=>[fieldName, fieldIds[index] ?? fieldValues[index]]
))
```

No mutation! Also this is perfect here because `filters` is const, so it can't be redefined. So without even looking at rest of the code you can guarantee that the above change won't break anything

Reply:
Resolved in https://github.com/specify/specify7/pull/4215/commits/e437cf48e953d7711923723c011d282c470b3f1e